docs/ref/signals.txt

```diff
@@ -219,6 +271,25 @@ Arguments sent with this signal:
 ``using``
     The database alias being used.
 
+Alternative to the post_delete signal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Unless the model you are targetting is provided by a library and cannot be swapped, a
+more robust and maintainable solution would be to override the model's
+:meth:`~django.db.models.Model.delete` method::
+
+    class Order(models.Model):
+        ...
+        @transaction.atomic
+        def delete(self, **kwargs):
+            result = super().delete(**kwargs)
+
+            # here goes the post_delete logic
+            self.product.stock += self.quantity
+            self.product.save(update_fields=("stock",))
+
+            return result
```

Comment:
Again I'm not sure this is a good example since it's not robust. Perhaps another `on_commit` email to say an order was deleted?

Reply:
Done :)